hasura/graphql-engine
# Replace idle GC with a custom GC thread

The current idle GC settings seem never to cause idle GC to trigger.
The changes here at least help memory usage to look more reasonable when
running certain benchmarks, and speculatively could partially fix some
memory leaks users have reported.

See `ourIdleGC` for details.

Referencing canonical memory issue #3388




### Description



### Changelog


### Affected components



### Related Issues




#3388 etc

### Solution and Design



### Steps to test and verify



Try some of the repros Tiru has been looking at. Observe memory drops somewhat after a few seconds after closing a benchmark, matching behavior of `-I0.3`

### Limitations, known bugs & workarounds



I thought i would use `-Iw` but it is actually not in GHC 8.10.1, and in any case actually doesn't get us all the way to what we want. See implementation.

### Server checklist


#### Catalog upgrade

Does this PR change Hasura Catalog version?

#### Metadata


Does this PR add a new Metadata feature?
  - Does `run_sql` auto manages the new metadata through schema diffing?
  - Does `run_sql` auto manages the definitions of metadata on renaming?
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?


#### GraphQL
   
   

#### Breaking changes


  1. Metadata API

     Existing `query` types:
     

  2. GraphQL API

     Schema Generation:
     
     
     

     Schema Resolve:-
     
     

  3. Logging
